### PR TITLE
Add EitherWriteable and use it for DoublePlaceholder in currency format

### PR DESCRIPTION
**Pull request by sffc**
Cleans up the currency formatter code a bit.

Part of #1441

### CODE REVIEW

File: utils/writeable/src/lib.rs [hunk 1]
```diff
@@ -319,6 +319,57 @@ pub trait Writeable {
     }
 }
 
+/// A [`Writeable`] that is either one type or another type.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[allow(clippy::exhaustive_enums)] // defined to be an enumeration over two types
+pub enum EitherWriteable<W0, W1> {
```
Comment (robertbastian) [inline utils/writeable/src/lib.rs:325]: `either::Either` is the standard Rust type to do this. Can we use that instead of defining our own?

Comment (sffc) [reply utils/writeable/src/lib.rs:325]: Fewer deps seems nice... we already have EitherYoke and EitherProvider defined in their respective crates

@Manishearth opinion?

Comment (Manishearth) [reply utils/writeable/src/lib.rs:325]: I'm okay with an optional dep here. I'm also somewhat open to moving EitherYoke to Either if possible.

> a "standard"

"standard Rust type" does not necessarily mean it is "a standard", these have two different implications. I'd say `either` _is_ the standard Rust type for this.

Comment (Manishearth) [reply utils/writeable/src/lib.rs:325]: In general when it comes to deps we already pull in in a couple places I'm less worried about them. I still want our utils to be simple but it's fine if it's optional deps for crates that are extremely commonly found in Rust.

**Comment by sffc** [comment]
Apparently `icu_datetime` already depends on `either` which makes me much less hesitant to add the impl that way

**Comment by Manishearth** [comment]
> Apparently `icu_datetime` already depends on `either` which makes me much less hesitant to add the impl that way

(yeah this is why I am also generally ok with introducing either in ICU4X for the larger deps)

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
@younies if you have any comments I can fix them in a follow-up. Turns out I need this PR for person names also, so I went ahead and merged it.

**Pull request merged**